tests/test_packages/test_connections/test_p2p_libp2p/test_public_dht.py

```diff
@@ -105,174 +85,72 @@ def delegate_uris_public_keys(request):
 class TestLibp2pConnectionPublicDHTRelay(BaseP2PLibp2pTest):
     """Test that public DHT's relay service is working properly"""
 
-    @pytest.mark.parametrize(
-        "maddrs", [PUBLIC_DHT_MADDRS, PUBLIC_STAGING_DHT_MADDRS], indirect=True
-    )
+    @pytest.mark.parametrize("maddrs", [PUBLIC_DHT_MADDRS], indirect=True)
     def test_connectivity(self, maddrs):
         """Test connectivity."""
-        for i, maddr in enumerate(maddrs):
-            temp_dir = os.path.join(self.t, f"dir_{i}")
-            connection = _make_libp2p_connection(
-                relay=False,
-                entry_peers=[maddr],
-                data_dir=temp_dir,
-            )
+
+        for maddr in maddrs:
+            connection = _make_libp2p_connection(relay=False, entry_peers=[maddr])
             multiplexer = Multiplexer([connection])
+            self.multiplexers.append(multiplexer)
             self.log_files.append(connection.node.log_file)
             multiplexer.connect()
+            assert_msg = f"Couldn't connect to public node {maddr}"
+            assert connection.is_connected is True, assert_msg
 
-            try:
-                assert (
-                    connection.is_connected is True
-                ), "Couldn't connect to public node {}".format(maddr)
-            except Exception:
-                raise
-            finally:
-                multiplexer.disconnect()
-
-    @pytest.mark.flaky(reruns=5, reruns_delay=5)
-    @pytest.mark.parametrize(
-        "maddrs", [PUBLIC_DHT_MADDRS, PUBLIC_STAGING_DHT_MADDRS], indirect=True
-    )
+    @pytest.mark.parametrize("maddrs", [PUBLIC_DHT_MADDRS], indirect=True)
     def test_communication_direct(self, maddrs):
-        """Test communication direct."""
-        for i, maddr in enumerate(maddrs):
-            multiplexers = []
```

Comment:
no more flaky markers used in ACN tests. I believe this was related to improper clean-up. Here we used a local variable `multiplexers` which we never disconnected. Now implemented to append to `cls.multiplexers` in base class on usage of`make_connection`, `make_client_connection` or `make_mailbox_connection`, which ensures disconnection occurs via `atexit` registration on class setup